torchchat/distributed/generate.py

```diff
@@ -0,0 +1,256 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+import asyncio
+import atexit
+import importlib.util
+import subprocess
+import threading
+from abc import abstractmethod
+from collections import deque
+from dataclasses import dataclass
+from functools import partial
+from os import environ
+from pathlib import Path
+from typing import List, Optional
+from uuid import uuid4
+
+import torch.multiprocessing as mp
+from torchchat.cli.builder import BuilderArgs, TokenizerArgs
+from torchchat.distributed.dist_run import NAME_TO_DISTRIBUTION_AND_DTYPE
+from torchchat.distributed.logging_utils import SingletonLogger
+
+logger = SingletonLogger.get_logger()
+
+
+def _setup_env(world_size: int, rank: int, target: callable, *args, **kwargs):
+    environ["MASTER_ADDR"] = "localhost"
+    environ["MASTER_PORT"] = "29500"
+    environ["RDZV_BACKEND"] = "c10d"
+    environ["WORLD_SIZE"] = str(world_size)
+    environ["RANK"] = str(rank)
+    environ["LOCALRANK"] = str(rank)
+
+    return target(*args, **kwargs)
+
+
+def _launch_distributed_inference(
+    model_name: str, builder_args: BuilderArgs, tokenizer_args: TokenizerArgs
+) -> tuple[List]:
+    # create programmatic elastic launch
```

Comment:
prefer to have one line doc strings as a rule for each function.  You kind of do this with this comment, but doc string is more informative that it applies for the whole function.

Reply:
Good practice! Added a one liner in each function